File: backend/app/schemas/classification.py
"""分类相关的Pydantic schemas"""
from pydantic import BaseModel, Field
from typing import List, Optional, Literal, Dict, Any


class ClassifyRequest(BaseModel):
    """分类请求"""
    text: str = Field(..., min_length=1, description="待分类的文本")


class ClassifyResponse(BaseModel):
    """分类响应"""
    type: Literal['schedule', 'memo'] = Field(..., description="分类类型")
    confidence: float = Field(..., ge=0, le=1, description="置信度(0-1)")
    extracted_data: Dict[str, Any] = Field(..., description="提取的结构化数据")
    needs_manual_selection: bool = Field(..., description="是否需要用户手动选择")
    
    class Config:
        from_attributes = True


class ManualClassifyRequest(BaseModel):
    """手动分类请求"""
    text: str = Field(..., min_length=1, description="文本内容")
    type: Literal['schedule', 'memo'] = Field(..., description="用户选择的类型")

File: frontend/src/api/classification.ts
/**
 * 文本分类API客户端
 */
import apiClient from './client'

export interface ClassifyRequest {
  text: string
}

export interface ClassifyResponse {
  type: 'schedule' | 'memo'
  confidence: number
  extracted_data: {
    date?: string
    time?: string
    description?: string
    summary?: string
    tags?: string[]
  }
  needs_manual_selection: boolean
}

export interface ManualClassifyRequest {
  text: string
  type: 'schedule' | 'memo'
}

/**
 * 自动分类文本
 */
export async function classifyText(text: string): Promise<ClassifyResponse> {
  const response = await apiClient.post<ClassifyResponse>('/classify', {
    text
  })
  return response.data
}

/**
 * 手动选择分类
 */
export async function manualClassify(text: string, type: 'schedule' | 'memo'): Promise<any> {
  const response = await apiClient.post('/classify/manual', {
    text,
    type
  })
  return response.data
}
